modules/marker/handle_periodic_operations.go

```diff
@@ -0,0 +1,72 @@
+package marker
+
+import (
+	"fmt"
+
+	"github.com/forbole/bdjuno/v3/modules/utils"
+	"github.com/forbole/bdjuno/v3/types"
+
+	markertypes "github.com/MonikaCat/provenance/x/marker/types"
+	"github.com/go-co-op/gocron"
+	"github.com/rs/zerolog/log"
+)
+
+// RegisterPeriodicOperations implements modules.PeriodicOperationsModule
+func (m *Module) RegisterPeriodicOperations(scheduler *gocron.Scheduler) error {
+	log.Debug().Str("module", "marker").Msg("setting up periodic tasks")
+
+	// Setup a cron job to run every hour
+	if _, err := scheduler.Every(1).Hour().Do(func() {
+		utils.WatchMethod(m.updateMarkersAccounts)
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateMarkersAccounts fetches from the REST APIs the latest markers
+// and saves them inside the database.
+func (m *Module) updateMarkersAccounts() error {
+	log.Debug().
+		Str("module", "marker").
+		Msg("getting markers data")
+
+	height, err := m.db.GetLastBlockHeight()
+	if err != nil {
+		return err
+	}
+
+	// Get the markers list
+	markersList, err := m.source.GetAllMarkers(height)
+	if err != nil {
+		return fmt.Errorf("error while getting markers list: %s", err)
+	}
+
+	var markers []types.MarkerAccount
+	for _, marker := range markersList {
+		var accountI markertypes.MarkerAccountI
+		err := m.cdc.UnpackAny(marker, &accountI)
+		if err != nil {
+			return err
+		}
+
+		var supply []types.MarkerSupply
+		// custom function GetSupplyValues
+		supplyDenom, supplyAmount := accountI.GetSupplyValues()
+		supply = append(supply, types.NewMarkerSupply(supplyDenom, supplyAmount.String()))
```

Comment:
it seems to me that the supply only has a single `sdk.Coin`, maybe the slice can be removed ?

Alternatively we can use `accountI.GetSupply()` method if we set the supply type to sdk.Coins

Reply:
Hey @huichiaotsou we couldn't use `accountI.GetSupply()` method coz it is throwing this error `panic: invalid denom: pm.pool.asset.1imruueldt6qxoizdsq0pb` - thats why I stored the `Supply` as `[]MarkerSupply` instead of the `sdk.Coins` type and I also added custom function `GetSupplyValues()` in provenance app to return the values without the error.

Reply:
ah I see :D